wgengine/magicsock/magicsock.go

```diff
@@ -1398,6 +1436,7 @@ Top:
 	} else {
 		ep = c.findEndpoint(ipp, addr)
 	}
+	noteRecvActivity(ep)
```

Comment:
If we unconditionally note activity here, why do we also need it in the above unlock+relock dance?

From context, I _think_ it's because we expect noteRecvActivity to update endpointOfDisco. Is that correct?

Reply:
The one above is `c.noteRecvActivity`, which is the lower level one in wgengine that ultimate does the CreateEndpoint.

This one down here is once we already have the endpoint, which tracks the activity of that endpoint.

But, yes, this second one is redundant in those rare cases. For performance reasons it's not worth caring about that double call in those rare cases, but for documentation/clarity, I added a bool to not call it twice redundantly.